provider/resource_couchbasecapella_database_user.go

```diff
@@ -0,0 +1,371 @@
+
+	// Only one of `buckets` or `all_bucket_access` can be specified in the terraform configuration file.
+	// The follwing checks handle errors where none or both are specified. If only one of `buckets` or
+	// `all_bucket_access` is specified, the bucket access roles will be set accordingly.
+	if !allBucketAccessExists && !bucketsExists {
+		return diag.Errorf("No bucket access roles specified")
+	}
+
+	if allBucketAccessExists && !bucketsExists {
+		allBucketAccess := couchbasecapella.BucketRoleTypes(d.Get("all_bucket_access").(string))
+		createDatabaseUserRequest.SetAllBucketsAccess(allBucketAccess)
+	}
+
+	if !allBucketAccessExists && bucketsExists {
+		buckets := expandBuckets(d)
+		createDatabaseUserRequest.SetBuckets(buckets)
+	}
+
+	if allBucketAccessExists && bucketsExists {
+		return diag.Errorf("Please specify only access for specific buckets or access for all buckets")
+	}
+
+	r, err := client.ClustersApi.ClustersCreateUser(auth, clusterId).CreateDatabaseUserRequest(createDatabaseUserRequest).Execute()
+	if r == nil {
+		return diag.Errorf("Pointer to database user create http.Response is nil")
+	}
+	if err != nil {
+		return manageErrors(err, *r, "Create Database User")
+	}
+
+	d.SetId(username)
+
+	return resourceCouchbaseCapellaDatabaseUserRead(ctx, d, meta)
+}
+
+// resourceCouchbaseCapellaDatabaseUserRead is responsible for reading a Couchbase
+// Capella database user using the Terraform resource data.
+func resourceCouchbaseCapellaDatabaseUserRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(*couchbasecapella.APIClient)
+	auth := getAuth(ctx)
+	clusterId := d.Get("cluster_id").(string)
+
+	// Check if the Cluster is inVPC to read the db users
+	// Managing buckets is not available for hosted clusters
+	_, _, err := client.ClustersApi.ClustersShow(auth, clusterId).Execute()
+
+	if err != nil {
+		// Check V3Cluster :: Need to be fixed in next versions
+		_, _, err3 := client.ClustersV3Api.ClustersV3show(auth, clusterId).Execute()
+		if err3 != nil {
+			return diag.FromErr(fmt.Errorf(ClusterProblemAccessing))
+		}
+		return diag.FromErr(fmt.Errorf(DatabaseUserHostedNotSupported))
+	}
+
+	// The current version of the Capella API doesn't support getting a singular
+	// database user. To obtain the database user, we need to iterate through
+	// the list of all database users and check if it exists. If the user
+	// is not present in the list of users and error is thrown.
+	timeout := time.NewTimer(time.Second * 120)
+	ticker := time.NewTicker(time.Second * 2)
+	for {
+		select {
+		case <-timeout.C:
+			username := d.Id()
+			d.SetId("")
+			return diag.Errorf("Error 404: Failed to find the username %s ", username)
+		case <-ticker.C:
+			users, _, err := client.ClustersApi.ClustersListUsers(auth, clusterId).Execute()
+			if err != nil {
+				return diag.FromErr(err)
+			}
+			for _, user := range users {
+				if user.Username == d.Id() {
+					return nil
+				}
```

Comment:
I've taken a look at implementing my suggestion and it seems quite complicated. It's gonna take some time to get this implemented like that, and since this is a in-vpc related resource that will soon be deprecated, I'm not sure if we want this to delay the release of the 0.1.